kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/actions/MultiFileConvertAction.java

```diff
@@ -0,0 +1,135 @@
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import org.kitodo.mediaserver.core.api.IAction;
+import org.kitodo.mediaserver.core.api.IConverter;
+import org.kitodo.mediaserver.core.api.IExtractor;
+import org.kitodo.mediaserver.core.api.IMetsReader;
+import org.kitodo.mediaserver.core.api.IReadResultParser;
+import org.kitodo.mediaserver.core.config.FileserverProperties;
+import org.kitodo.mediaserver.core.config.MetsProperties;
+import org.kitodo.mediaserver.core.conversion.FileEntry;
+import org.kitodo.mediaserver.core.db.entities.Work;
+import org.kitodo.mediaserver.core.exceptions.ConversionException;
+import org.kitodo.mediaserver.core.util.MediaServerUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+
+/**
+ * An action to convert all master images to one PDF file.
+ */
+public class MultiFileConvertAction implements IAction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MultiFileConvertAction.class);
+
+    private FileserverProperties fileserverProperties;
+    private MetsProperties metsProperties;
+    private IMetsReader metsReader;
+    private IMetsReader fullPdfMetsReader;
+    private IReadResultParser readResultParser;
+    private Map<String, IConverter> converters = new HashMap<>();
+    private IExtractor patternExtractor;
+    private MediaServerUtils mediaServerUtils;
+
+    @Autowired
+    public void setFileserverProperties(FileserverProperties fileserverProperties) {
+        this.fileserverProperties = fileserverProperties;
+    }
+
+    @Autowired
+    public void setMetsProperties(MetsProperties metsProperties) {
+        this.metsProperties = metsProperties;
+    }
+
+    public void setMetsReader(IMetsReader metsReader) {
+        this.metsReader = metsReader;
+    }
+
+    public void setReadResultParser(IReadResultParser readResultParser) {
+        this.readResultParser = readResultParser;
+    }
+
+    public void setFullPdfReader(IMetsReader fullPdfMetsReader) {
+        this.fullPdfMetsReader = fullPdfMetsReader;
+    }
+
+    public Map<String, IConverter> getConverters() {
+        return converters;
+    }
+
+    public void setPatternExtractor(IExtractor patternExtractor) {
+        this.patternExtractor = patternExtractor;
+    }
+
+    @Autowired
+    public void setMediaServerUtils(MediaServerUtils mediaServerUtils) {
+        this.mediaServerUtils = mediaServerUtils;
+    }
+
+    /**
+     * Performes a conversion of a given work.
+     *
+     * @param work the work entity
+     * @param parameter conversion parameter
+     * @return an input stream with the conversion or null
+     * @throws Exception if anything goes wrong
+     */
+    public InputStream perform(Work work, Map<String, String> parameter) throws Exception {
+
+        LOGGER.info("Preproducing full PDF for work " + work.getId());
```

Comment:
It's true, that this should preferrably be used for preproduction, but this class shouldn't decide (or even know) that